From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Tue, 10 Jul 2018 00:00:01 +0200
Subject: include/feeds.mk: rework generation of opkg distfeeds.conf

Allow enabling/commenting/disabling each feed individually by using a
tristate config symbol.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

diff --git a/include/feeds.mk b/include/feeds.mk
index c9ffa95a368a72fd43a7e688b7ee1b274ade2916..3e0801e656bfa4d14ccd22db23a059d46cdc558c 100644
--- a/include/feeds.mk
+++ b/include/feeds.mk
@@ -10,8 +10,6 @@
 
 FEEDS_INSTALLED:=$(notdir $(wildcard $(TOPDIR)/package/feeds/*))
 FEEDS_AVAILABLE:=$(sort $(FEEDS_INSTALLED) $(shell $(SCRIPT_DIR)/feeds list -n))
-FEEDS_ENABLED:=$(foreach feed,$(FEEDS_AVAILABLE),$(if $(CONFIG_FEED_$(feed)),$(feed)))
-FEEDS_DISABLED:=$(filter-out $(FEEDS_ENABLED),$(FEEDS_AVAILABLE))
 
 PACKAGE_SUBDIRS=$(PACKAGE_DIR)
 ifneq ($(CONFIG_PER_FEED_REPO),)
@@ -35,10 +33,11 @@ endef
 # 1: destination file
 define FeedSourcesAppend
 ( \
-  echo "src/gz %d_core %U/targets/%S/packages"; \
+  echo 'src/gz %d_core %U/targets/%S/packages'; \
+  echo 'src/gz %d_base %U/packages/%A/base'; \
   $(strip $(if $(CONFIG_PER_FEED_REPO), \
-	$(foreach feed,base $(FEEDS_ENABLED),echo "src/gz %d_$(feed) %U/packages/%A/$(feed)";) \
-	$(if $(CONFIG_PER_FEED_REPO_ADD_DISABLED), \
-		$(foreach feed,$(FEEDS_DISABLED),echo "$(if $(CONFIG_PER_FEED_REPO_ADD_COMMENTED),# )src/gz %d_$(feed) %U/packages/%A/$(feed)";)))) \
+	$(foreach feed,$(FEEDS_AVAILABLE), \
+		$(if $(CONFIG_FEED_$(feed)), \
+			echo '$(if $(filter m,$(CONFIG_FEED_$(feed))),# )src/gz %d_$(feed) %U/packages/%A/$(feed)';)))) \
 ) >> $(1)
 endef
diff --git a/package/base-files/Makefile b/package/base-files/Makefile
index ef29798d5de855e35d178cb02f356bb3e77a9bd0..4842a45a41afac0f21b63ee2f1a71a443b234141 100644
--- a/package/base-files/Makefile
+++ b/package/base-files/Makefile
@@ -25,8 +25,6 @@ PKG_CONFIG_DEPENDS += \
 	CONFIG_NAND_SUPPORT \
 	CONFIG_CLEAN_IPKG \
 	CONFIG_PER_FEED_REPO \
-	CONFIG_PER_FEED_REPO_ADD_DISABLED \
-	CONFIG_PER_FEED_REPO_ADD_COMMENTED \
 	$(foreach feed,$(FEEDS_AVAILABLE),CONFIG_FEED_$(feed))
 
 include $(INCLUDE_DIR)/package.mk
diff --git a/package/base-files/image-config.in b/package/base-files/image-config.in
index 5ee2d3e4e320d337a0fb0f46ae4649ae847cd6c2..cee8cd54e1239a3dd424fa86df8fac3eaa1d31b8 100644
--- a/package/base-files/image-config.in
+++ b/package/base-files/image-config.in
@@ -268,18 +268,4 @@ menuconfig PER_FEED_REPO
 		If set, a separate repository is generated within bin/*/packages/
 		for the core packages and each enabled feed.
 
-	config PER_FEED_REPO_ADD_DISABLED
-		bool "Add available but not enabled feeds to opkg.conf"
-		default y
-		depends on PER_FEED_REPO
-		help
-		  Add not installed or disabled feeds from feeds.conf to opkg.conf.
-
-	config PER_FEED_REPO_ADD_COMMENTED
-		bool "Comment out not enabled feeds"
-		default !BUILDBOT
-		depends on PER_FEED_REPO && PER_FEED_REPO_ADD_DISABLED
-		help
-		  Add not enabled feeds as commented out source lines to opkg.conf.
-
 source "tmp/.config-feeds.in"
diff --git a/scripts/feeds b/scripts/feeds
index b29e1d5c353d23c8f0afc81fb0f1ec37e1f2a521..304ef6cbafd141ff98d6a8afe5de5b371a4cb2a7 100755
--- a/scripts/feeds
+++ b/scripts/feeds
@@ -824,11 +824,12 @@ sub feed_config() {
 		my $installed = (-f "feeds/$feed->[1].index");
 
 		printf "\tconfig FEED_%s\n", $feed->[1];
-		printf "\t\tbool \"Enable feed %s\"\n", $feed->[1];
+		printf "\t\ttristate \"Enable feed %s\"\n", $feed->[1];
 		printf "\t\tdepends on PER_FEED_REPO\n";
 		printf "\t\tdefault y\n" if $installed;
 		printf "\t\thelp\n";
-		printf "\t\t Enable the \\\"%s\\\" feed at %s.\n", $feed->[1], $feed->[2][0];
+		printf "\t\t Enable the \\\"%s\\\" feed in opkg distfeeds.conf.\n", $feed->[1];
+		printf "\t\t Say M to add the feed commented out.\n";
 		printf "\n";
 	}
 
